sdks/java/core/src/main/java/org/apache/beam/sdk/io/ClassLoaderFileSystem.java

```diff
@@ -0,0 +1,154 @@
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.io.fs.CreateOptions;
+import org.apache.beam.sdk.io.fs.MatchResult;
+import org.apache.beam.sdk.io.fs.ResolveOptions;
+import org.apache.beam.sdk.io.fs.ResourceId;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+
+/** A read-only {@link FileSystem} implementation looking up resources using a ClassLoader. */
+public class ClassLoaderFileSystem extends FileSystem<ClassLoaderFileSystem.ClassLoaderResourceId> {
+
+  public static final String SCHEMA = "classpath";
+  private static final String PREFIX = SCHEMA + "://";
+
+  ClassLoaderFileSystem() {}
+
+  @Override
+  protected List<MatchResult> match(List<String> specs) throws IOException {
+    throw new UnsupportedOperationException("Un-globable filesystem.");
+  }
+
+  @Override
+  protected WritableByteChannel create(
+      ClassLoaderResourceId resourceId, CreateOptions createOptions) throws IOException {
+    throw new UnsupportedOperationException("Read-only filesystem.");
+  }
+
+  @Override
+  protected ReadableByteChannel open(ClassLoaderResourceId resourceId) throws IOException {
+    ClassLoader classLoader = getClass().getClassLoader();
+    InputStream inputStream =
+        classLoader.getResourceAsStream(resourceId.path.substring(PREFIX.length()));
+    if (inputStream == null) {
+      throw new IOException("Unable to load " + resourceId.path + " with " + classLoader);
+    }
+    return Channels.newChannel(inputStream);
+  }
+
+  @Override
+  protected void copy(
+      List<ClassLoaderResourceId> srcResourceIds, List<ClassLoaderResourceId> destResourceIds)
+      throws IOException {
+    throw new UnsupportedOperationException("Read-only filesystem.");
+  }
+
+  @Override
+  protected void rename(
+      List<ClassLoaderResourceId> srcResourceIds, List<ClassLoaderResourceId> destResourceIds)
+      throws IOException {
+    throw new UnsupportedOperationException("Read-only filesystem.");
+  }
+
+  @Override
+  protected void delete(Collection<ClassLoaderResourceId> resourceIds) throws IOException {
+    throw new UnsupportedOperationException("Read-only filesystem.");
+  }
+
+  @Override
+  protected ClassLoaderResourceId matchNewResource(String path, boolean isDirectory) {
+    return new ClassLoaderResourceId(path);
+  }
+
+  @Override
+  protected String getScheme() {
+    return SCHEMA;
+  }
+
+  public static class ClassLoaderResourceId implements ResourceId {
+
+    private final String path;
+
+    private ClassLoaderResourceId(String path) {
+      checkArgument(path.startsWith(PREFIX), path);
+      this.path = path;
+    }
+
+    @Override
+    public ResourceId resolve(String other, ResolveOptions resolveOptions) {
```

Comment:
Can we add a couple trivial unit tests as sanity checks / documentation for `resolve` and `getCurrentDirectory`?

Reply:
The documentation is in the super classes, but I added a test.

Reply:
Thanks!